pkg/resources/deployments/deployment.go

```diff
@@ -26,8 +26,107 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
+// Returns the Mcs environment variables set in configuration.
+// If a user specifies a set of secrets for the MCS Environment
+func mcsEnvironmentVars(mi *miniov1beta1.MinIOInstance) []corev1.EnvVar {
+	envVars := make([]corev1.EnvVar, 0)
+	if mi.HasMcsSecret() {
+		var secretName string
+		secretName = mi.Spec.Mcs.McsSecret.Name
+		envVars = append(envVars, corev1.EnvVar{
+			Name: "MCS_HMAC_JWT_SECRET",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: secretName,
+					},
+					Key: "mcshmacjwt",
+				},
+			},
+		}, corev1.EnvVar{
+			Name: "MCS_PBKDF_PASSPHRASE",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: secretName,
+					},
+					Key: "mcspbkdfpassphrase",
+				},
+			},
+		}, corev1.EnvVar{
+			Name: "MCS_PBKDF_SALT",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: secretName,
+					},
+					Key: "mcspbkdfsalt",
+				},
+			},
+		}, corev1.EnvVar{
+			Name: "MCS_SECRET_KEY",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: secretName,
+					},
+					Key: "mcssecretkey",
+				},
+			},
+		}, corev1.EnvVar{
```

Comment:
let's sort this properly with MCS_MINIO_SERVER as the first element and then subsequently added SECRET_KEY, ACCESS_KEY and then add the rest